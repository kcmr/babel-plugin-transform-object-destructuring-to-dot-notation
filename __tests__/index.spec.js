const babel = require('@babel/core');
const plugin = require('..');

function transformFixture(fixture, options = {}) {
  const { code } = babel.transform(fixture, {
    plugins: [[plugin, options]],
  });

  return code;
}

it('transforms properties from destructured object to dot notation', () => {
  const code = `
    const { foo, bar } = myObject;
    console.log(foo);
    console.log(bar);
  `;

  const result = transformFixture(code, { name: 'myObject' });
  expect(result).toMatchSnapshot();
});

it('does not transform properties from different scope', () => {
  const code = `
    const { foo, bar, fiz } = myObject;

    function baz() {
      const { foo } = anotherObject;
      console.log(foo);
      console.log(fiz);

      const bar = anotherObject.bar;
      console.log(bar);
    }
  `;

  const result = transformFixture(code, { name: 'myObject' });
  expect(result).toMatchSnapshot();
});

it('does not transform code without name option', () => {
  const code = 'const { foo } = myObject';
  const result = transformFixture(code);

  expect(result).toMatchSnapshot();
});
