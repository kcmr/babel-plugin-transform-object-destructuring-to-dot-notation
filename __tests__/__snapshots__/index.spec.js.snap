// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not transform code without name option 1`] = `
"const {
  foo
} = myObject;"
`;

exports[`does not transform properties from different scope 1`] = `
"function baz() {
  const {
    foo
  } = anotherObject;
  console.log(foo);
  console.log(myObject.fiz);
  const bar = anotherObject.bar;
  console.log(bar);
}"
`;

exports[`transforms properties from destructured object to dot notation 1`] = `
"console.log(myObject.foo);
console.log(myObject.bar);"
`;
